library/kani/src/futures.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, RawWaker, RawWakerVTable, Waker},
+};
+
+/// Polls the future in a loop until completion (a very simple executor)
```

Comment:
Explain a little more what this does and doesn't do

Reply:
As for what it does, I think the comment explains that already. What do you think is missing?

As for what it doesn't do: clever executors will only poll the future again if the waker that was passed in the previous `poll` call is invoked. But Kani cannot just sleep/be idle for a certain amount of time, so busy-looping is the only option. If we're doing that, wakers don't help, so we don't need to support this. Being able to ignore most of the `waker` stuff also makes things easier for CBMC.

Reply:
I've added some explanation to the documentation for that function.